### PR TITLE
Initialise all layers on file load.

### PR DESCRIPTION
**Pull request by alekamca**
This pull request is to resolve this issue: https://github.com/AnimalLogic/AL_USDMaya/issues/26

### CODE REVIEW

File: lib/AL_USDMaya/AL/usdmaya/Global.cpp [hunk 1]
```diff
@@ -61,7 +61,36 @@ static void preFileOpen(void*)
 {
   TF_DEBUG(ALUSDMAYA_EVENTS).Msg("preFileOpen\n");
 }
+//----------------------------------------------------------------------------------------------------------------------
+static void initialiseChildAndSublayers(nodes::Layer& layer, nodes::ProxyShape* proxy)
+{
+  // initialise self
+  MPlug plug = layer.nameOnLoadPlug();
+  MString path = plug.asString();
+  if(path.length() && path.substring(0, 3) != "anon")
```
Comment (dbaz) [inline lib/AL_USDMaya/AL/usdmaya/Global.cpp]: I noticed we had this substring check before. Maybe we can do it better this time, try use SdfLayer::IsAnonymousLayerIdentifier to determine if it's anonymous

File: lib/AL_USDMaya/AL/usdmaya/Global.cpp [hunk 2]
```diff
@@ -61,7 +61,36 @@ static void preFileOpen(void*)
 {
   TF_DEBUG(ALUSDMAYA_EVENTS).Msg("preFileOpen\n");
 }
+//----------------------------------------------------------------------------------------------------------------------
+static void initialiseChildAndSublayers(nodes::Layer& layer, nodes::ProxyShape* proxy)
```
Comment (dbaz) [inline lib/AL_USDMaya/AL/usdmaya/Global.cpp]: Move initialiseChildAndSublayers function to be in Layer.h/cpp

File: lib/AL_USDMaya/AL/usdmaya/Global.cpp [hunk 3]
```diff
@@ -86,29 +115,19 @@ static void postFileOpen(void*)
...
-        {
-          SdfLayerHandle layer = SdfLayer::FindOrOpen(path.asChar());
-          LAYER_HANDLE_CHECK(layer);
-          layerPtr->setLayerAndClearAttribute(layer);
-        }
-        else
-        {
+          layer->init(proxy, stage->GetSessionLayer());
+
+          // initialise and fixup each child and sublayer
```
Comment (dbaz) [inline lib/AL_USDMaya/AL/usdmaya/Global.cpp]: Stick these sublayer and childlayer loops into a member function into Layer.h/cpp and call if from here

**Comment by alekamca** [comment]
Thank you for reviewing--all of the comments make sense. I will try to address them soon.

**Comment by dbaz** [comment]
Hey @alekamca as you already know, https://github.com/AnimalLogic/AL_USDMaya/pull/53 changes how we are dealing with layers in AL_USDMaya and affects this PR.

What is the way forward for this code review? can that change be made compatible with how you are using the layers at your studio?

I'm really digging for more information on how exactly you guys are using the Layers in AL_USDMaya, and what you are trying to do.

Let us know.

**Comment by alekamca** [comment]
@dbaz I think it's safe to close this PR and try to figure out how to make our uresolved path stuff work with the new layer manager. Looking more into how the Tf error notification works, I think that is probably the better path forward anyways.

**Pull request closed without merging**